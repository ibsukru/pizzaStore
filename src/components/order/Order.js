import React from 'react'
import Pizza from '../pizza'
import Basket from '../basket'
import Toppings from '../toppings'

const Order = ({
  onToggleBasket,
  basket,
  pizza,
  data: { loading, pizzaSizes }
}) => {
  if (loading) return <h1>Loading...</h1>

  if (basket && basket.show) {
    return <Basket basket={basket} />
  }

  const orders = basket && basket.orders

  const basketMarkup =
    orders &&
    orders.length > 0 &&
    <div className='basket-actions'>
      <a className='basket-actions__toggle' onClick={onToggleBasket}>
        Go to basket ({orders.length})
      </a>
    </div>

  if (pizza) {
    return <Toppings pizza={pizza} />
  }

  return (
    <div>
      <h1>Welcome to your pizza store.</h1>
      <Pizza sizes={pizzaSizes} />
      {basketMarkup}
    </div>
  )
}

export default Order
