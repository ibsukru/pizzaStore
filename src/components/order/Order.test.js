/* eslint-env jest */
import React from 'react'
import Order from './Order'
import { create } from 'react-test-renderer'

[
  {
    onToggleBasket: f => f,
    data: { loading: true, pizzaSizes: null }
  }
].forEach(props => {
  test(`With props ${JSON.stringify(props)} Should match snapshot...`, () => {
    expect(create(<Order {...props} />).toJSON()).toMatchSnapshot()
  })
})
