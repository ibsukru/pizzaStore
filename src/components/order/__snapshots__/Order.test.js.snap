// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With props {"data":{"loading":true,"pizzaSizes":null}} Should match snapshot... 1`] = `
<h1>
  Loading...
</h1>
`;
